fix: prevent empty screen instead of dashboard; log to file

krrprr opened the pull request:
Fixes the following issues:
- when the `setup` command of xud-launcher results in error, an error message and `Retry` button is displayed. If the detailed error message cannot be found from the response, the user is recommended to see the details from the application logs;
- on navigating to the dashboard, it is checked if connection to the xud-docker-api can be established. If not, the user is redirected to the home page;
- all errors and info messages are now logged to the file instead of console.

The release can be found [here](https://github.com/ExchangeUnion/xud-ui/releases/tag/v1.2.1-alpha.1).

raladev reviewed (commented):
In case of proxy container stopping/crashing I immediately getting start page without any error no matter what it was Create flow or Connect. There is no significant difference between this behavior and black screen appearing - in both cases im getting not expected screen without any instructions.

Can we add hint for that case?

For Create flow (local env) - connection with proxy lost, please check status of proxy container and the logs.
For Connect flow (remove env) - connection lost, please check accessibility of {URL with getinfo}

krrprr commented:
> In case of proxy container stopping/crashing I immediately getting start page without any error no matter what it was Create flow or Connect. There is no significant difference between this behavior and black screen appearing - in both cases im getting not expected screen without any instructions.
> 
> Can we add hint for that case?
> 
> For Create flow (local env) - connection with proxy lost, please check status of proxy container and the logs.
> For Connect flow (remove env) - connection lost, please check accessibility of {URL with getinfo}

Used the same page that is currently shown on macOS and Linux when connection lost. Just modified the text and button and removed `How to set up environment` link. A more advanced solution will be implemented with https://github.com/ExchangeUnion/xud-ui/issues/144.

Release can be found [here](https://github.com/ExchangeUnion/xud-ui/releases/tag/v1.2.1-alpha.4).
